h/static/scripts/auth.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * @ngdoc service
+ * @name  auth
+ *
+ * @description
+ * The 'auth' service exposes authorization helpers for other components.
+ */
+
+Auth.$inject = ['jwtHelper'];
+function Auth  ( jwtHelper ) {
+  this.userid = function userid(identity) {
+    try {
+      if (jwtHelper.isTokenExpired(identity)) {
+        return null;
+      } else {
+        var payload = jwtHelper.decodeToken(identity);
```

Comment:
Eww to an API that let's you decode an expired token without massive flashing red lights, but hey -- you work with what you're given.